velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -151,25 +148,33 @@ class S3ReadFile final : public ReadFile {
   // bytes.
   void preadInternal(uint64_t offset, uint64_t length, char* position) const {
     // Read the desired range of bytes.
-    Aws::S3::Model::GetObjectRequest request;
-    Aws::S3::Model::GetObjectResult result;
-
-    request.SetBucket(awsString(bucket_));
-    request.SetKey(awsString(key_));
-    std::stringstream ss;
-    ss << "bytes=" << offset << "-" << offset + length - 1;
-    request.SetRange(awsString(ss.str()));
-    request.SetResponseStreamFactory(
-        AwsWriteableStreamFactory(position, length));
-    auto outcome = client_->GetObject(request);
-    VELOX_CHECK_AWS_OUTCOME(outcome, "Failed to get S3 object", bucket_, key_);
+    Aws::Transfer::TransferManagerConfiguration transfer_config(
+        executor_.get());
+    transfer_config.s3Client = client_;
+    auto transfer_manager =
+        Aws::Transfer::TransferManager::Create(transfer_config);
+    Aws::Utils::Stream::PreallocatedStreamBuf streamBuffer(
+        (unsigned char*)position, length);
+    auto downloadHandle =
+        transfer_manager->DownloadFile(bucket_, key_, offset, length, [&]() {
+          return Aws::New<UnderlyingStreamWrapper>("TestTag", &streamBuffer);
+        });
+    downloadHandle->WaitUntilFinished();
```

Comment:
Does WaitUntilFinished() wait indefinitely, or times out after some time? 
If it is a timeout, please mention the value as a comment.

Reply:
Wait's forever...good catch.  Is there a timeout value that should be observed or should this be a new configuration value?

Reply:
default to 5s, and expose the value as a configuration.
@majetideepak - what do you suggest?

Reply:
Can we end up in a case when `5s` is not enough? Ideally, we want to avoid configs.

Reply:
Anything I do to add a timeout value here will be a busy wait. There is no TransferHandle.Wait() with a timeout. I can test completion of all parts but I don't think I can recommend a busy loop. I agree with @majetideepak, I added one for TransferManager max threads at a request but I always thought 25 threads max was a good number.

Reply:
It might be good to log some message somewhere if these requests take a long time.